test/extended/operators/olm.go

```diff
@@ -37,12 +36,72 @@ var _ = g.Describe("[Feature:Platform] OLM should", func() {
 			e2e.Logf("Successfully list %s", olmResources[i])
 		})
 	}
+
+	// OCP-22070
+
+	g.It("[ocp-22070]support grpc sourcetype", func() {
+		catalogsourceYF := "https://raw.githubusercontent.com/jianzhangbjz/v3-testfiles/v4.1/olm/catalogsource/image-catalogsource.yaml"
+		subscriptionYF := "https://raw.githubusercontent.com/jianzhangbjz/v3-testfiles/v4.1/olm/subscription/ocp-22070-sub.yaml"
+
+		createResourceByYAML(oc, catalogsourceYF)
+		time.Sleep(30 * time.Second)
+		catalogCmd := []string{"get", "packagemanifest", "-n", "openshift-operator-lifecycle-manager", "etcd"}
+		checkResource(oc, catalogCmd)
+
+		createResourceByYAML(oc, subscriptionYF)
+		time.Sleep(60 * time.Second)
+		getCmds := map[int][]string{
+			0: {"get", "sub", "-n", "openshift-operators", "test-operator"},
```

Comment:
Is there any reason for this to be a map?

Reply:
I assume so that order is enforced; go slices are iterated randomly

Reply:
Yes, thanks! I'd like to check them in order.

Reply:
@njhale Since `checkResource` receive the `slice` as arguments, I intended to put these `slices` into together. So I used `map`. I will sort this `map`, does it make sense? Or any suggestions? Thanks!

Reply:
@jianzhangbjz You can have a slice of slices, e.g.:

```go 
getCmds := [][]string{
  {"get", "sub", "-n", "openshift-operators", "test-operator"},
  {"get", "csv", "-n", "openshift-operators", "etcdoperator.v0.9.2"},
  {"get", "pods", "-n", "openshift-operators", "etcd-operator"},
}
for _, c := range getCmds {
  checkResource(oc, c)
}
```

Reply:
@jianzhangbjz adding to @ecordell 's suggestion, you can also create a struct to hold the arguments:

```go
type resource struct {
    kind string
    namespace string
    name string
}

get := []resource{
    {"sub", "openshift-operators", "test-operator"},
    {"csv", "openshift-operators", "etcdoperator.v0.9.2"},
    {"pods", "openshift-operators", "etcd-operator"},
}

// ...
func checkResource(oc *exutil.CLI, res resource) {
    output, err := oc.AsAdmin().WithoutNamespace().Run("get").Args(fmt.Sprintf("%s -n %s %s", res.kind, res.namespace, res.name)).Output()
    // ...    
}
```